pam/gdm/conversation_test.go

```diff
@@ -0,0 +1,107 @@
+package gdm
+
+import (
+	"testing"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/pam/pam_test"
+)
+
+func TestSendToGdm(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(pam_test.MaybeDoLeakCheck)
+
+	testCases := map[string]struct {
+		value []byte
+
+		wantReturn []byte
+		wantError  error
+	}{
+		"JSON null data can be sent and received": {
+			value: []byte(`null`),
+		},
+
+		"JSON number can be sent and received": {
+			value: []byte(`1.5`),
+		},
+
+		"Single char is sent and received as string": {
+			value: []byte(`"m"`),
+		},
+
+		"JSON null is returned": {
+			value:      []byte(`"give me 🚫"`),
+			wantReturn: []byte("null"),
+		},
+
+		"Utf-8 data is sent and returned": {
+			value:      []byte(`"give me 🍕"`),
+			wantReturn: []byte(`"😋"`),
+		},
+
+		// Error cases
+		"Error on empty data": {
+			value:     []byte{},
+			wantError: ErrInvalidJSON,
+		},
+
+		"Error on nil data": {
+			value:     nil,
+			wantError: ErrInvalidJSON,
+		},
+
+		"Error with empty data returned": {
+			value:      []byte(`"give me 🗑‼"`),
+			wantReturn: []byte{},
+			wantError:  ErrInvalidJSON,
+		},
+
+		"Error with nil data returned": {
+			value:      []byte(`"give me 🚫"`),
+			wantReturn: []byte(nil),
+			wantError:  ErrInvalidJSON,
+		},
+	}
+
+	for name, tc := range testCases {
```

Comment:
New code, old comment about the spacing between the test cases (also the spacing between the test cases and the `for` loop)...